service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsController.java

```diff
@@ -77,16 +80,17 @@
     public CallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         caseDetails.getData().remove(SELECTED_HEARING_ID);
-        caseDetails.getData().remove("hearingOption");
-        caseDetails.getData().remove("hearingReListOption");
 
         CaseData caseData = getCaseData(caseDetails);
 
+        boolean isFirstHearing = isEmpty(caseData.getAllHearings());
+
         if (caseData.getAllocatedJudge() != null) {
             caseDetails.getData().put("judgeAndLegalAdvisor", setAllocatedJudgeLabel(caseData.getAllocatedJudge()));
         }
 
-        caseDetails.getData().put(FIRST_HEARING_FLAG, (isEmpty(caseData.getHearingDetails()) ? YES : NO).getValue());
+        caseDetails.getData().put(FIRST_HEARING_FLAG, YesNo.from(isFirstHearing).getValue());
+        caseDetails.getData().put(PRE_ATTENDANCE, isFirstHearing ? DEFAULT_PRE_ATTENDANCE : null);
```

Comment:
Are you sure you want to put a null value here (instead of not adding any values in case isFirstHearing is false)?

Reply:
yeah, shorter code base, but if it is hard to read, then yes - changed